soroban-env-host/src/native_contract/token/storage_types.rs

```diff
@@ -12,6 +12,7 @@ pub struct AllowanceDataKey {
 pub struct BalanceValue {
     pub amount: i128,
     pub authorized: bool,
```

Comment:
This might be a preliminary optimization, but maybe we should introduce a u32 flags field instead? I think even with 2 bools it will be more optimal in terms of ledger space and also easier to extend going forward.

Reply:
I would prefer to keep it as is for now because it's much easier to understand. Also I don't think u32 would be more optimal in terms of ledger space unless we had over 4 bools right?

Reply:
Even if we optimize the structs to exclude field names, every bool still comes with +4 bytes for tag, so u32 is 4 byte tag + 4 byte payload, 2 bools is 2x4 byte tag (not sure if value takes any space, so maybe they're at least equal). But I'm mostly concerned about extensibility here.

Reply:
Yeah the point on extensibility is valid. I think we should avoid adding any more flags, but using a u32 instead wouldn't hurt (other than making the code a little more complex). What do you think about merging the PR and then I can open an issue for this?

Reply:
Sure, that makes sense.